h2o-py/tests/testdir_misc/pyunit_job_progress_custom_widget.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import print_function
+from builtins import range
+import sys
+
+
+sys.path.insert(1,"../../")
+import h2o
+from h2o.job import H2OJob
+from h2o.utils.progressbar import ProgressBar, ProgressBarWidget, PBWBar, PBWString, RenderResult
+from tests import pyunit_utils as pu
+
+job_request_counter = 0
+
+
+class MyWidget(ProgressBarWidget):
+
+    out = pu.StringIO()
+
+    def render(self, progress, width=None, status=None):
+        if status != 'init':
+            self.out.write("%s\n" % progress)
+        if status is not None:
+            self.out.write("%s\n" % status)
+        return RenderResult()
```

Comment:
@AakashKumarNain you can do pretty mutch anything you want in this `render` method.
By default, the `RenderResult` is sent to`sys.stdout`, as it's probably not what you need in Wave, you could update some Wave progress bar, and return an empty `RenderResult` like in this example.